src/App.tsx

```diff
@@ -1,56 +1,102 @@
-import React from 'react';
+import React, { useState, useCallback } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [selectPerson, setSelectPerson] = useState<Person | null>(null);
+  const [inputValue, setInputValue] = useState<string>('');
+  const [focusInput, setFocusInput] = useState<boolean>(false);
+
+  const people = peopleFromServer;
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const newValue: string = event.target.value;
+
+    setInputValue(newValue);
+
+    if (!newValue) {
+      setSelectPerson(null);
+    }
+  };
+
+  const handlePersonClick = useCallback((selectedPerson: Person) => {
+    setSelectPerson(selectedPerson);
+    setInputValue(selectedPerson.name);
+    setFocusInput(false);
+  }, []);
+
+  const filteredPerson = people.filter((person) => {
+    return person.name.toLowerCase().includes(inputValue.toLowerCase());
+  });
+  const title = selectPerson ? `${selectPerson.name} (${selectPerson.born} - ${selectPerson.died})` : 'No matching suggestions';
```

Comment:
Consider renaming the variable 'title' to something more descriptive like 'selectedPersonTitle'.